esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -698,6 +698,57 @@ bool WsWuInfo::legacyHasSubGraphTimings()
     return false;
 }
 
+void WsWuInfo::getGraphs(WUGraphType graphType, IArrayOf<IEspECLGraph>& graphs)
+{
+    SCMStringBuffer runningGraph;
+    WUGraphIDType id;
+
+    WUState st = cw->getState();
+    bool running = (!(st==WUStateFailed || st==WUStateAborted || st==WUStateCompleted) && cw->getRunningGraph(runningGraph,id));
+
+    Owned<IConstWUGraphMetaIterator> it = &cw->getGraphsMeta(GraphTypeAny);
```

Comment:
Why are you fetching all graphs then filtering yourself, rather than passing the type to getGraphsMeta?